test/unit/common/test_index2d.cpp

```diff
@@ -137,17 +142,32 @@ TYPED_TEST(Index2DTest, Print) {
 }
 
 TYPED_TEST(Index2DTest, ComputeLinearIndex) {
-  std::array<dlaf::common::Ordering, 2> orderings{dlaf::common::Ordering::RowMajor,
-                                                  dlaf::common::Ordering::ColumnMajor};
+  using dlaf::common::Ordering;
+  std::array<Ordering, 2> orderings{Ordering::RowMajor, Ordering::ColumnMajor};
+
+  const Index2D<TypeParam> index(13, 26);
+  std::vector<Size2D<TypeParam>> configs{{
+      Size2D<TypeParam>{13, 26},  // out-of-rows & out-of-cols
+      Size2D<TypeParam>{13, 39},  // out-of-rows
+      Size2D<TypeParam>{26, 26},  // out-of-cols
+  }};
 
   for (const auto& ordering : orderings) {
-    EXPECT_EQ(-1, computeLinearIndex(ordering, Index2D<TypeParam>{13, 26},
-                                     {13, 26}));  // out-of-rows & out-of-cols
-    EXPECT_EQ(-1, computeLinearIndex(ordering, Index2D<TypeParam>{13, 26}, {13, 39}));  // out-of-rows
-    EXPECT_EQ(-1, computeLinearIndex(ordering, Index2D<TypeParam>{13, 26}, {26, 26}));  // out-of-cols
+    for (const Size2D<TypeParam>& size : configs) {
+      EXPECT_DEATH(computeLinearIndex(ordering, index, {size.rows(), size.cols()}), "[ERROR]");
+      EXPECT_DEATH(computeLinearIndex(ordering, index, Size2D<TypeParam>(size)), "[ERROR]");
```

Comment:
Are those tests really needed?  Moreover this test shouldn't fail in release mode (ASSERT_MODERATE).

Note that death tests are quite expensive, therefore we should use them only in special cases.

Reply:
Good point. IMHO they are useful because they tests the guards and related preconditions. I know that you think that since they are preconditions, they should not be tested. But I would like to test that the "service" of the guards works as expected.

In case, we would decide to maintain them, I have two proposal:
1. forcefully define the flags `DLAF_ASSERT_...` that we need at the beginning of the file;
2. enable this thests just when the flags are enabled on build (i.e. `#ifdef DLAF_ASSERT*`)

Personally, I would go for the second.

Reply:
Good point. IMHO they are useful because they tests the guards and related preconditions. I know that you think that since they are preconditions, they should not be tested. But I would like to test that the "service" of the guards works as expected.

In case, we would decide to maintain them, I have two proposal:
1. forcefully define the flags `DLAF_ASSERT*` that we need at the beginning of the file;
2. enable this the tests just when the flags are enabled on build (i.e. `#ifdef DLAF_ASSERT*`)

Personally, I would go for the second.

Reply:
Second options is better.
First option may create unexpected behaviour with functions compiled in the library